sample-wasm-chaincode/chaincode_example02/rust/src/lib.rs

```diff
@@ -1,300 +1,285 @@
+    let result_key_len = get_parameter(0, first_account.as_ptr());
+    first_account_name_len = usize::try_from(result_key_len).unwrap();
 
+    //get to account
+    let result_key_len = get_parameter(1, second_account.as_ptr());
+    second_account_len = usize::try_from(result_key_len).unwrap();
 
-    // Perform the execution
-    let a_val = a_val - txn_amount;
-    let b_val = b_val + txn_amount;
+    //get amount to transfer
+    let txn_amount_ptr = [0; 24];
+    let result_key_len = get_parameter(2, txn_amount_ptr.as_ptr());
+    //Validate and convert amount to integer
+    let txn_amount_length = usize::try_from(result_key_len).unwrap();
+    let the_bytes = &txn_amount_ptr[0..txn_amount_length];
+    let the_string = str::from_utf8(the_bytes).expect("not UTF-8");
+    txn_amount = the_string.parse().expect("not a number");
+
+
+    //Get asset balance of A and B
+    let from_account_balance: u64;
+    let to_account_balance: u64;
 
-    let ttemp = format!("{}{}{}{}", "Aval = ", a_val," Bval = ",b_val);
-    let t = ttemp.as_bytes();
+    //get from account balance
+    let get_result = [0; 24];
+    let result_key_len = unsafe { __get_state(first_account.as_ptr(), first_account_name_len, get_result.as_ptr()) };
+    if result_key_len == -1 {
+        let error_msg = "ERROR! from account not found".as_bytes();
+        print(error_msg.as_ptr(), error_msg.len());
+        return_result(error_msg.as_ptr(), error_msg.len());
+        return -1;
+    }
 
-    // Write the state back to the ledger
-    unsafe {
-        __print(t.as_ptr(), t.len());
+    // convert byte array -> string -> integer
+    let account_bal_length = usize::try_from(result_key_len).unwrap();
+    let the_bytes = &get_result[0..account_bal_length];
+    let the_string = str::from_utf8(the_bytes).expect("not UTF-8");
+    from_account_balance = the_string.parse().expect("not a number");
+
+    //get to account balance
+    let get_result = [0; 24];
+    let result_key_len = unsafe { __get_state(second_account.as_ptr(), second_account_len, get_result.as_ptr()) };
+    if result_key_len == -1 {
+        let error_msg = "ERROR! to account not found".as_bytes();
+        print(error_msg.as_ptr(), error_msg.len());
+        return_result(error_msg.as_ptr(), error_msg.len());
+        return -1;
+    }
 
-        let stemp =a_val.to_string();
-        let s = stemp.as_bytes();
-        let result1 = __put_state(a.as_ptr(), a_length,s.as_ptr(), s.len());
+    let account_bal_length = usize::try_from(result_key_len).unwrap();
+    let the_bytes = &get_result[0..account_bal_length];
+    let the_string = str::from_utf8(the_bytes).expect("not UTF-8");
+    to_account_balance = the_string.parse().expect("not a number");
 
-        if result1==-1 {
-            return -1;
-        }
+    // Perform the execution
+    let from_account_balance_updated = from_account_balance - txn_amount;
+    let to_account_balance_updated = to_account_balance + txn_amount;
```

Comment:
I might have missed the line, but should we check the balance before performing the execution? This is just an example so I am okay as is regardless of having the check or not.

Reply:
I agree, will add the check